### PR TITLE
Functionnal tests for api media binary

### PR DESCRIPTION
**Pull request by srascar**
Must be merge after this PR on SonataMediaBundle https://github.com/sonata-project/SonataMediaBundle/pull/705

### CODE REVIEW

File: features/api/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -31,11 +31,17 @@ class FeatureContext extends BehatContext
...
+     */
+    private $filesPath;
+
+    /**
      * {@inheritdoc}
      */
     public function __construct(array $parameters)
     {
         $this->baseUrl = $parameters['base_url'];
+        $this->filesPath = $parameters['files_path'];
```
Comment (Bladrak) [inline features/api/bootstrap/FeatureContext.php]: Indentation

File: features/api/bootstrap/FeatureContext.php [hunk 2]
```diff
@@ -459,5 +465,65 @@ public function theResponseShouldBeABinary()
...
+                $path = $fullPath;
+            }
+        }
+
+        if (($content = file_get_contents($path)) === false) {
+            throw new \Exception(sprintf('Unable to get the content of the binary %s', $path));
+        }
+
+        $headers = array(
+            'Content-type'=> 'application/octet-stream',
```
Comment (Bladrak) [inline features/api/bootstrap/FeatureContext.php]: Indent

File: features/api/bootstrap/FeatureContext.php [hunk 3]
```diff
@@ -459,5 +465,65 @@ public function theResponseShouldBeABinary()
             throw new Exception(sprintf('Response Content-Disposition header not attachment: "%s"', $response->getHeader('Content-Disposition')));
         }
     }
+
+    /**
+     * @When /^(?:I )?send a ([A-Z]+) request to "([^"]*)" with the binary "([^"]*)"$/
+     */
+    public function iSendARequestWithTheBinary($method, $url, $path)
```
Comment (Bladrak) [inline features/api/bootstrap/FeatureContext.php]: Missing PHPDoc

File: features/api/bootstrap/FeatureContext.php [hunk 4]
```diff
@@ -459,5 +465,65 @@ public function theResponseShouldBeABinary()
...
+        );
+
+        $this->sendRequestWithIdentifiers($method, $url, $headers, $content);
+    }
+
+
+    /**
+     * @When /^(?:I )?send a ([A-Z]+) request containing identifier to "([^"]*)"$/
+     */
+    public function iSendAGetRequestContainingIdentifierTo($method, $url)
```
Comment (Bladrak) [inline features/api/bootstrap/FeatureContext.php]: PHPDoc

File: features/api/media/workflow.feature [hunk 5]
```diff
@@ -77,4 +77,42 @@ Feature: Check the API for MediaBundle
...
+    And store the XML response identifier as "image_media_id"
+
+    When I send a GET request to "/api/media/media/<image_media_id>.xml" using last identifier:
+    Then the response code should be 200
+    And response should contain "xml" object
+    And response should contain "My image media"
+    And response should contain "missing_binary_content"
+
+
+    When I send a PUT request to "/api/media/media/<image_media_id>/binary/content.json" with the binary "features/fixtures/sonata.jpg"
```
Comment (Bladrak) [inline features/api/media/workflow.feature]: Why is this a request in JSON format where other requests are in XML format?

**Comment by rande** [comment]
@srascar is this PR ready ?

**Pull request merged**